On some versions of AppKit, the programmatic Paste menu title shows up as "…"

whsieh opened the pull request:
#### 553ca1c5d0c25e616a74f03fa47aa1a5ee82072d
<pre>
On some versions of AppKit, the programmatic Paste menu title shows up as &quot;…&quot;
<a href="https://bugs.webkit.org/show_bug.cgi?id=252753">https://bugs.webkit.org/show_bug.cgi?id=252753</a>
rdar://105781726

Reviewed by Aditya Keerthi.

Simplify some of this logic for presenting and positioning the DOM paste popover when requesting
programmatic pasteboard access. Instead of using a confinement rect to force the popover into the
correct position on screen (which, on certain versions of macOS, now causes the menu item to be
clipped), we should just supply the correct mouse event location when presenting the menu item in
the first place by converting the mouse location from screen to window coordinates.

* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::requestDOMPasteAccess):
(-[WKDOMPasteMenuDelegate confinementRectForMenu:onScreen:]): Deleted.

Canonical link: <a href="https://commits.webkit.org/260689@main">https://commits.webkit.org/260689@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1c136b03355e14b623a00f53220ab975a8d9f806

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/109069 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/18151 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/41891 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/604 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/118336 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/19610 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/9453 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/101310 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/114829 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/14700 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/97932 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/42838 "Found 2 new test failures: webgl/2.0.y/conformance2/extensions/promoted-extensions-in-shaders.html, webgl/2.0.y/conformance2/extensions/webgl-multi-draw-instanced-base-vertex-base-instance.html (failure)") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/96672 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/29585 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/84589 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/10933 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/30931 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/11672 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/7861 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/17046 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/50530 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7382 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/13278 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder reviewed (approved):
Do we need to preserve the old implementation for older OSes?

whsieh committed:
On some versions of AppKit, the programmatic Paste menu title shows up as "…"
https://bugs.webkit.org/show_bug.cgi?id=252753
rdar://105781726

Reviewed by Aditya Keerthi.

Simplify some of this logic for presenting and positioning the DOM paste popover when requesting
programmatic pasteboard access. Instead of using a confinement rect to force the popover into the
correct position on screen (which, on certain versions of macOS, now causes the menu item to be
clipped), we should just supply the correct mouse event location when presenting the menu item in
the first place by converting the mouse location from screen to window coordinates.

* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::requestDOMPasteAccess):
(-[WKDOMPasteMenuDelegate confinementRectForMenu:onScreen:]): Deleted.

Canonical link: https://commits.webkit.org/260689@main
